src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -593,8 +580,11 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
 
     /* All empty lines or lines with only white spaces will be skipped.
      * The invalidity info returned are in HTML format.
+     *
+     * @param studentList is built to be used in enrollStudents(String, String, boolean)
      */
-    private List<String> getInvalidityInfoInEnrollLines(String lines, String courseId) throws EnrollException {
+    private List<String> getInvalidityInfoInEnrollLines(String lines, String courseId, List<StudentAttributes> studentList)
```

Comment:
If anything, the method signature and name does not signify the need for a list. Why would a method to get invalidity info in enroll lines need a student list to be passed in? Granted that you may get some performance boost (which is your objective, I believe), this is the kind of hack that will sacrifice readability and maintainability. As a proof, look at the "hacks" you need to add to the tests, and how those "dummy" lists are unused afterwards.

If I understand your intention correctly, the best middle ground is:
```java
public List<StudentAttributes> createAndValidateStudents(String lines, String courseId) throws EnrollException {
    List<StudentAttributes> studentList = new ArrayList<StudentAttributes>();
    ...
    if (!invalidityInfo.isEmpty()) {
        throw new EnrollException(StringHelper.toString(invalidityInfo, "<br>"));
    }
    return studentList;
}
```
Notice that the method does exactly what you want and does not sacrifice readability either.